compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/selection/TextFieldSelectionManager.kt

```diff
@@ -311,6 +311,9 @@ internal class TextFieldSelectionManager(
                     isStartHandle = false,
```

Comment:
Is it right to make snapshot after selection? The issue is about non-working undo after we change the value (cut it with ctrl/cmd-X). Maybe we should make snapshot only after such kind of actions?

Reply:
So make a snapshot _before_ the command? That's an option indeed. Will move the line tomorrow.